src/scripts/safari/Safari.ts

```diff
@@ -284,6 +291,46 @@ class Safari {
         }
     }
 
+    public static spawnPokemonCheck() {
+        this.steps++;
+        if (this.steps % 10 === 0 && Math.round(Math.random())) {
+            this.spawnRandomPokemon();
+        }
+    }
+
+    public static despawnPokemonCheck() {
+        this.pokemonGrid = this.pokemonGrid.map(column => column.map(pokemon => {
+            if (pokemon && --pokemon.steps <= 0) {
+                pokemon.element?.remove();
+                pokemon = null;
+            }
+            return pokemon;
+        }));
+    }
+
+    private static spawnRandomPokemon() {
+        const y = Math.floor(Math.random() * this.sizeY());
+        const x = Math.floor(Math.random() * this.sizeX());
+        if (!this.canMove(x, y) || (x == this.playerXY.x && y == this.playerXY.y) || this.pokemonGrid[y][x]) {
+            return;
+        }
+        const pokemon = SafariPokemon.random();
+        
+        const offset = {
+            top: 32 * y - 8,
+            left: 32 * x - 4,
+        };
+        pokemon.element = document.createElement('div');
+        pokemon.element.classList.value = `pokemonSprite walkDown ${pokemon.shiny ? 'shiny' : ''}`;
+        pokemon.element.style.backgroundImage = `url('assets/images/dynamic-background/pokemon/${pokemon.id.toString().padStart(3, '0')}${pokemon.shiny ? 's' : ''}.png')`;
+        pokemon.element.style.top = `${offset.top}px`;
+        pokemon.element.style.left = `${offset.left}px`;
+        document.getElementById('safariBody').appendChild(pokemon.element);
```

Comment:
This looks like something we should be doing with knockout

Reply:
I believe all of the safari minigame (other than the modal+wrapping content and battle) is being rendered with straight JS right now, since a decent amount of code is for setting positions and animation frames. I'm torn on this because I actually prefer it as pure JS, and porting to knockout would be a bit of work, but it is also inconsistent with the rest of the game. I'm content leaving this outside of knockout for now though

Reply:
I've moved it to knockout bindings 👍